snapcraft/internal/meta/_desktop.py

```diff
@@ -38,9 +39,9 @@ def __init__(
                 filename, "does not exist (defined in the app {!r})".format(name)
             )
 
-    def parse_and_reformat(self):
+    def parse_and_reformat(self, extracted_metadata: _metadata.ExtractedMetadata):
         self._parser = configparser.ConfigParser(interpolation=None)
-        self._parser.optionxform = str
+        self._parser.optionxform = str  # type: ignore
```

Comment:
why is this type ignored?

Reply:
Otherwise mypy complains about "Incompatible types in assignment", see https://github.com/python/mypy/issues/5062

However, I don't remember seeing this error before, and mypy remains at version 0.590.

Reply:
Can you just add a comment above that line, something like

    # mypy type checking ignored, see python/mypy#5062